tfx_bsl/beam/run_inference.py

```diff
@@ -522,22 +730,24 @@ def __init__(
     self._session = None
     self._io_tensor_spec = None
 
-  def setup(self):
-    """Load the model.
-
-    Note that worker may crash if exception is thrown in setup due
-    to b/139207285.
-    """
+    self._validate_model()
 
-    super(_BaseBatchSavedModelDoFn, self).setup()
-    self._tags = _get_tags(self._inference_spec_type)
+    self._tags = _get_tags(inference_spec_type)
     self._io_tensor_spec = self._pre_process()
 
     if self._has_tpu_tag():
       # TODO(b/131873699): Support TPU inference.
       raise ValueError('TPU inference is not supported yet.')
     self._session = self._load_model()
 
+  def _validate_model(self):
+    """Optional subclass model validation hook.
```

Comment:
Why is this optional?

Reply:
it's optional in the sense that subclasses can choose whether to implement it. For instance `_BatchMultiInferenceDoFn` does not do any extra model validation. If this wording is confusing, we could just write: `"Subclass model validation hook."`